tikv/kv.go

```diff
@@ -165,16 +171,35 @@ func (s *KVStore) Go(f func()) error {
 	return s.gP.Run(f)
 }
 
-// UpdateTxnSafePointCache updates the cached txn safe point, which is used for safety check of data access
+// updateTxnSafePointCache updates the cached txn safe point, which is used for safety check of data access
 // operations to prevent accessing GC-ed inconsistent data.
-func (s *KVStore) UpdateTxnSafePointCache(txnSafePoint uint64, now time.Time) {
+func (s *KVStore) updateTxnSafePointCache(txnSafePoint uint64, now time.Time) {
 	s.gcStateCacheMu.Lock()
 	defer s.gcStateCacheMu.Unlock()
 
 	s.gcStateCacheMu.lastCacheTime = now
 	s.gcStateCacheMu.cachedTxnSafePoint = txnSafePoint
 }
 
+func (s *KVStore) loadTxnSafePoint(ctx context.Context) (uint64, error) {
+	if s.gcStatesAPIUnavailable.Load() {
+		logutil.Logger(ctx).Warn("GC states API is not available, which is possibly caused by cluster version < 9.0. Txn safe point updating will be fallen back to direct etcd reading.")
```

Comment:
Would this introduce a lot of logs if is called much times by the tools?

Reply:
🤔
How about changing it to Debug level and only print the warning once?

Reply:
Do we have metrics or clues to verify the requests coming from new or fallback old API? Or do we need such diagnosing information?

Reply:
I'll try to add a metric